examples/docker2singularity/docker2singularity.sh

```diff
@@ -101,21 +101,52 @@ $SUDOCMD singularity copy $new_container_name group /etc/group
 rm singularity.json
 $SUDOCMD rm grouphost
 $SUDOCMD rm group
+$SUDOCMD singularity bootstrap $new_container_name
 
 ################################################################################
 ### SINGULARITY RUN SCRIPT #####################################################
 ################################################################################
 
 CMD=$($SUDOCMD docker inspect --format='{{json .Config.Cmd}}' $image)
 # Remove quotes and braces
-CMD=`echo "${CMD//\"/}" | sed 's/\[//g' | sed 's/\]//g'`
-if [[ $CMD != none ]]; then
-  echo '#!/bin/sh'
-  (IFS='[],'; echo $CMD)
-fi > singularity
-chmod +x singularity
-sudo singularity copy $new_container_name singularity /
-rm singularity
+CMD=`echo "${CMD//\"/}" | sed 's/\[/\/bin\/sh -c /g' | sed 's/\]//g'`
+
+ENTRYPOINT=$($SUDOCMD docker inspect --format='{{json .Config.Entrypoint}}' $image)
+# Remove quotes and braces
+ENTRYPOINT=`echo "${ENTRYPOINT//\"/}" | sed 's/\[/\/bin\/sh -c /g' | sed 's/\]//g'`
+
+echo '#!/bin/sh' > /tmp/singularity
+if [[ $ENTRYPOINT != "null" ]]; then
+    if [[ $CMD != "null" ]]; then
+        echo $ENTRYPOINT $CMD >> /tmp/singularity;
+    else
+        echo $ENTRYPOINT >> /tmp/singularity;
+    fi
+else
+    echo $CMD >> /tmp/singularity;
+fi
+
+chmod +x /tmp/singularity
+$SUDOCMD singularity copy $new_container_name /tmp/singularity /
+rm /tmp/singularity
+
+################################################################################
+### SINGULARITY ENVIRONMET #####################################################
+################################################################################
+
+docker run --entrypoint env scitran/fsl-fast > /tmp/docker_environment
```

Comment:
and this is funny because I did the same thing when I first wrote this script with ubuntu:latest :)